### PR TITLE
Implements starknet.go Class RPC endpoints for builder

### LINKED ISSUE #9: Implement `starknet.go` support for contract class endpoints
Add autogenerated `starknet.go` code for the following RPC endpoints

- `starknet_getClass`
- `starknet_getClassHashAt`
- `starknet_getClassAt`

**Additional Context:**

- https://github.com/NethermindEth/rpc-request-builder/issues/1
- https://github.com/NethermindEth/rpc-request-builder/blob/main/src/api/rpcspec/methods.ts

### PR DESCRIPTION
**Pull request by dhirajs0**
support for starknet.go code for the following RPC endpoints

starknet_getClass
starknet_getClassHashAt
starknet_getClassAt

Closes #9

### CODE REVIEW

File: src/api/rpcspec/methods.ts [hunk 1]
```diff
@@ -7,6 +7,24 @@ const provider = new RpcProvider({
...
+const STARKNET_GO_PREFIX = `
+import (
+	"context"
+	"fmt"
+	"log"
+	"github.com/NethermindEth/starknet.go/rpc"
+	"github.com/NethermindEth/starknet.go/utils"
+)
+func Class() {
+	rpcUrl := "https://free-rpc.nethermind.io/mainnet-juno/?apikey=GGfMLnf5icG5MM68u2mV8gXj83bx2uC9KowlL2yoKQSM71b3"
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:19]:
```suggestion
	rpcUrl := "https://free-rpc.nethermind.io/mainnet-juno/"
```

File: src/api/rpcspec/methods.ts [hunk 2]
```diff
@@ -362,7 +401,18 @@ const ReadMethods = [
...
+    if err != nil {
+      log.Fatal(err)
+    }
+  
+    result, err := provider.ClassAt(context.Background(), rpc.BlockID{Tag: "latest"}, contractAddress)
+    if err != nil {
+      log.Fatal(err)
+    }
+    
+    fmt.Println("ClassOutput : ", result)
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:414]:
```suggestion
    fmt.Println("ClassOutput: ", result)
```

File: src/api/rpcspec/methods.ts [hunk 3]
```diff
@@ -347,7 +375,18 @@ const ReadMethods = [
...
+    if err != nil {
+      log.Fatal(err)
+    }
+  
+    result, err := provider.ClassHashAt(context.Background(), rpc.BlockID{Tag: "latest"}, contractAddress)
+    if err != nil {
+      log.Fatal(err)
+    }
+    
+    fmt.Println("ClassHash is :", result)
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:388]:
```suggestion
    fmt.Println("ClassHash:", result)
```

File: src/api/rpcspec/methods.ts [hunk 4]
```diff
@@ -332,7 +350,17 @@ const ReadMethods = [
...
+    starknetGo: `${STARKNET_GO_PREFIX}classHash, err := utils.HexToFelt("0x3131fa018d520a037686ce3efddeab8f28895662f019ca3ca18a626650f7d1e")
+    if err != nil {
+      log.Fatal(err)
+    }
+  
+    result, err := provider.Class(context.Background(), rpc.BlockID{Tag: "latest"}, classHash)
+    if err != nil {
+      log.Fatal(err)
+    }
+    fmt.Println("ClassOutput : ", result)
```
Comment (jelilat) [inline src/api/rpcspec/methods.ts:362]:
```suggestion
    fmt.Println("ClassOutput: ", result)
```

**Review by jelilat — CHANGES_REQUESTED** [review]
Please make the changes and resolve the conflicts and it should be GTM.

**Comment by dhirajs0** [comment]
> Please make the changes and resolve the conflicts and it should be GTM.

Made the suggested changes.

**Comment by jelilat** [comment]
Thank you for your contribution @dhirajs0

**Comment by AryanGodara** [comment]
@jelilat I had a doubt in these implementations. Since this is an RPC builder, and the requests are editable.
image
Initially, the code is correct, according to the fields like `latest` or `blockHash`.

But, as soon as I edit any input field, the code restructures, and is no longer correct. So, the `configuration of the request` isn't really working. I noticed this in most examples other than javascript, ie for Go and Rust
Eg.:
image
After changing classHash, the `hextoFelt line disappears`
image
after changing block number, the input parameter changes completely.

**Comment by jelilat** [comment]
@AryanGodara This happens because the only acceptable block tag is "latest" or "pending". If you edit the block tag to pending, it should be properly updated. 

For example, this is what it looks like if I change the tag to "pending"

**Comment by jelilat** [comment]
@AryanGodara This happens because the only acceptable block tag is "latest" or "pending". If you edit the block tag to pending, it should be properly updated. 

For example, this is what it looks like if I change the tag to "pending"

Screenshot 2024-03-06 at 15 04 09

**Pull request merged**